src/mainloop.c

```diff
@@ -165,5 +218,8 @@ int main()
         }
     }
 
+    free(cfg);
+    free(events);
```

Comment:
Do not touch `free(events)` and `free(request)`. Minimize the functional changes.